src/components/constraints/ConstraintsPanel.svelte

```diff
@@ -2,56 +2,321 @@
 
 <script lang="ts">
   import { base } from '$app/paths';
-  import { checkConstraintsStatus, constraints } from '../../stores/constraints';
-  import type { Constraint } from '../../types/constraint';
+  import ChecklistIcon from '@nasa-jpl/stellar/icons/checklist.svg?component';
+  import FilterIcon from '@nasa-jpl/stellar/icons/filter.svg?component';
+  import PlanLeftArrow from '@nasa-jpl/stellar/icons/plan_with_left_arrow.svg?component';
+  import PlanRightArrow from '@nasa-jpl/stellar/icons/plan_with_right_arrow.svg?component';
+  import VisibleHideIcon from '@nasa-jpl/stellar/icons/visible_hide.svg?component';
+  import VisibleShowIcon from '@nasa-jpl/stellar/icons/visible_show.svg?component';
+  import {
+    checkConstraintsStatus,
+    constraintViolations,
+    constraintViolationsMap,
+    constraintVisibilityMap,
+    constraints,
+    setAllConstraintsVisible,
+    setConstraintVisibility,
+  } from '../../stores/constraints';
+  import { field } from '../../stores/form';
+  import { plan, viewTimeRange } from '../../stores/plan';
+  import type { Constraint, ConstraintViolation } from '../../types/constraint';
+  import type { FieldStore } from '../../types/form';
   import type { ViewGridSection } from '../../types/view';
   import effects from '../../utilities/effects';
+  import { getDoyTime, getUnixEpochTime } from '../../utilities/time';
+  import { tooltip } from '../../utilities/tooltip';
+  import { required, timestamp } from '../../utilities/validators';
+  import CollapsibleListControls from '../CollapsibleListControls.svelte';
+  import DatePickerField from '../form/DatePickerField.svelte';
   import GridMenu from '../menus/GridMenu.svelte';
-  import CssGrid from '../ui/CssGrid.svelte';
+  import DatePickerActionButton from '../ui/DatePicker/DatePickerActionButton.svelte';
   import Panel from '../ui/Panel.svelte';
   import PanelHeaderActionButton from '../ui/PanelHeaderActionButton.svelte';
   import PanelHeaderActions from '../ui/PanelHeaderActions.svelte';
   import ConstraintListItem from './ConstraintListItem.svelte';
 
   export let gridSection: ViewGridSection;
 
+  let showAll: boolean = true;
   let filterText: string = '';
   let filteredConstraints: Constraint[] = [];
+  let endTimeDoy: string;
+  let endTimeDoyField: FieldStore<string>;
+  let startTimeDoy: string;
+  let startTimeDoyField: FieldStore<string>;
+  let showFilters: boolean = false;
+  let showConstraintsWithNoViolations: boolean = true;
+  let constraintViolationMap: Record<Constraint['name'], ConstraintViolation> = {}; // TODO ugh messy, store version maps name -> ConstraintViolation[], can you really have a list of these things? Model vs plan?
```

Comment:
@camargo any thoughts on this? Right now the UI is looking for a single violation entry per constraint. Although maybe this will change if/when that Aerie tweak happens to refactor the constraint violation API response?

Reply:
Yeah let's revisit in https://github.com/NASA-AMMOS/aerie-ui/issues/624